packages/mobile/src/components/fields/TextField.tsx

```diff
@@ -43,19 +45,44 @@ export const TextField = (props: TextFieldProps) => {
     styles: stylesProp,
     id,
     onChangeText,
+    error: errorProp,
+    debouncedValidationMs,
     ...other
   } = props
 
   const styles = useStyles()
-  const [{ value, onChange, onBlur }, { touched, error }] = useField(name)
+  const [
+    { value, onChange, onBlur },
+    { touched, error: errorMessage },
+    { setTouched }
+  ] = useField(name)
+
+  const { validateField, submitCount } = useFormikContext()
+
+  const debouncedValidateField = useDebouncedCallback(
+    (field: string) => validateField(field),
+    [validateField],
+    500
```

Comment:
I think you meant to use the `debouncedValidationMs` here?

```suggestion
    debouncedValidationMs
```

Reply:
yes, nice